### DISCUSSION TITLE
Zettlr snap

### DISCUSSION THREAD

**Discussion by ghost** [Zettlr/Zettlr#3376] [Ideas]
Hello folks,

Igor from the snap team in Canonical here.

I hope this is the right place to ask the following question - would you be interested in creating a Zettlr snap and then have it published in the Snap Store? I believe the technical effort of actually packaging the app as a snap would not be too difficult, and of course the snap team at Canonical can help if required.

I also think you would see good traction with the audience and wider reach in the Linux community without having to worry about the maintenance complexities of individual distributions and their specific dependencies. But then, of course, the proof is in the pudding, your own experience with the ecosystem, and your own requirements. Hopefully, we can talk about that in this discussion.

Looking forward to hearing your thoughts (good or bad) :)

**Comment by nathanlesage** [comment]
Hey Igor,

In general I'm fine with Zettlr being wherever people want it — as long as the maintenance efforts are manageable (i.e. either someone else manages the build process as in the AUR repo or we can just add another build target) and there are no bigger issues like incompatibilities etc (on some systems, Electron itself is a weird dependency and causes problems if multiple Electron based apps use different Electron versions — something I haven't understood until today).

I just did a quick check, and it would be possible to have Electron builder spit out a Snap package. Then the question would be how to get it onto the snap store? Right from the CI pipeline we could either upload the package directly somewhere, call a webhook, or we just publish all the files on GitHub and have a webhook on the release event. But in this regard you're definitely the expert!

**Reply by ghost** [reply]
Hey,

There are several options here:

1. Build the snap with Electron builder and then manually upload to the snap store.

2. Use Electron builder snap/publish options:
https://www.electron.build/configuration/publish

Specifically, take a look at SnapStoreOptions, and what can be passed on.

3. Use Electron builder with a "Custom" publish provider:
https://github.com/electron-userland/electron-builder/issues/3261
https://gist.github.com/layerssss/5d7b69c0f8c6e54e8b501e6e0fe36186

You can then combine this with the Snap Store API:
http://api.snapcraft.io/docs/info.html

4. Build the snap with Electron builder but publish to GH instead of the Snap Store.

Set snap.publish to:
{
provider: "generic",
url: "github url"
}

Then, use the GHA action-publish to publish to the Snap Store:
https://github.com/snapcore/action-publish

Let me know what you think.
